src/netbsd.rs

```diff
@@ -56,8 +56,19 @@ fn init() -> *mut c_void {
         const POLYFILL: GetRandomFn = polyfill_using_kern_arand;
         ptr = POLYFILL as *mut c_void;
     }
-    GETRANDOM.store(ptr, Ordering::Release);
-    ptr
+    // TODO(MSRV 1.64): Consider using `Ordering::Release` instead of
+    // `Ordering::AcqRel`; see https://github.com/matklad/once_cell/issues/220
+    // and https://github.com/rust-lang/rust/pull/98383.
+    match GETRANDOM.compare_exchange(ptr::null_mut(), ptr, Ordering::AcqRel, Ordering::Acquire) {
+        // We won the race; `GETRANDOM` now has the value `ptr`.
+        Ok(_) => ptr,
+        // We lost the race; another thread stored a different value. We always
+        // use the first value stored so that we only ever use one
+        // implementation. This usually won't matter but in theory it could
+        // matter if a sandbox or antivirus or something is causing dlsym to
+        // act in a non-idempotent way.
+        Err(previously_stored) => previously_stored,
```

Comment:
(Sorry for my previous comment, I left it only after a passing glance on your changes, so I deleted it)

Firstly, this concern looks too theoretic to me. Secondly, even if `dlsym` behaves is such erratic way, does it matter? Any pointer returned by `dlsym` should be valid, so overwriting older value should not cause any issues in practice and we do not do any cleanup, so it's not like we worry about leaking resources.

Reply:
It is much easier to make this change than it is for us to think about when/if/how we should care about whether dlsym is idempotent. The mental model is that we want something like `OnceLock` so we might as well have it act like OnceLock.

If it were possible that the performance of this store mattered, then I think we'd have something to consider. But since the performance of this store doesn't matter, I don't see any value in the current approach.

Reply:
`compare_exchange` is more complex than simple `store`. Even with a theoretical non-idempotent `dlsym` the latter should work without issues, so I don't see why we need to introduce this complexity. Mimicking `OnceLock` is not a good reason in my opinion.

Reply:
@josephlr WDYT? I am inclined to leave the current code as-is.

Reply:
My thought is that our Atomic synchronization/initialization routines (`LazyBool` and the weak function pointer in this PR) should have similar semantics.

Either all should use a "first wins" approach (this PR) or a "last wins" approach (the existing code).

I agree that "first wins" is easier to reason about. `std` seems to use a mix of approches:
  - "first wins": [`lazy_box`](https://github.com/rust-lang/rust/blob/a9c8887c7d548abc6c3e87f7d6fa02a0e95880bd/library/std/src/sys_common/lazy_box.rs#L56)
  - "last wins": [`DlsymWeak`](https://github.com/rust-lang/rust/blob/a9c8887c7d548abc6c3e87f7d6fa02a0e95880bd/library/std/src/sys/pal/unix/weak.rs#L131)

I personally agree that the "first wins" is a little easier to reason about, but given that libstd uses the "last wins" approch, I would keep things as-is. If we later add other synchronization primitives which use `compare_exchange`, we can switch this as well.